interfaces/IMonitor.h

```diff
@@ -0,0 +1,89 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#pragma once
+
+#include "Module.h"
+
+// @stubgen:include <com/IIteratorType.h>
+
+namespace Thunder {
+namespace Exchange {
+    /* @json 1.0.0 */
+    struct EXTERNAL IMonitor : virtual public Core::IUnknown {
+        enum { ID = ID_MONITOR };
+        struct Measurement {
+            uint64_t min /* @brief Minimal value measured */;
+            uint64_t max /* @brief Maximal value measured */;
+            uint64_t average /* @brief Average of all measurements */;
+            uint64_t last /* @brief Last measured value */;
+        };
+        struct Statistics {
+            Measurement resident /* @brief Resident memory measurement */;
+            Measurement allocated /* @brief Allocated memory measurement */;
+            Measurement shared /* @brief Shared memory measurement */;
+            Measurement process /* @brief Processes measurement */;
+            bool operational /* @brief Whether the service is up and running */;
+            uint32_t count /* @brief Number of measurements */;
+        };
+        struct RestartInfo {
+            uint8_t limit /* @brief Maximum number or restarts to be attempted */;
+            uint16_t window /* @brief Time period (in seconds) within which failures must happen for the limit to be considered crossed */;
+        };
+
+        using IStringIterator = RPC::IIteratorType<string, RPC::ID_STRINGITERATOR>;
+
+        // @event
+        struct EXTERNAL INotification : virtual public Core::IUnknown {
+            enum { ID = ID_MONITOR_NOTIFICATION };
+            enum action : uint8_t {
+                DEACTIVATE,
+                ACTIVATE,
+                RESTARTING_IS_STOPPED,
+            };
+            // @brief Signals an action taken by the Monitor
+            // @param callsign: Callsign of the service the Monitor acted upon
+            // @param action: The action executed by the Monitor on a service. One of: Activate, Deactivate, StoppedRestarting
+            // @param reason: A message describing the reason the action was taken
+            virtual void Actionconst string& callsign, const action value, const string& reason) = 0;
+        };
+
+        // Pushing notifications to interested sinks
+        virtual Core::hresult Register(INotification* notification) = 0;
+        virtual Core::hresult Unregister(INotification* notification) = 0;
+
+        // @property
+        // @brief RestartLimits: Set/Get new restart limits for a plugin
```

Comment:
Again this will look odd in the documentation, ie. no need to repeat the method name.

Reply:
done